contents/blog/clickhouse-vs-elasticsearch.md

```diff
@@ -0,0 +1,228 @@
+
+Others, like us, also use ClickHouse to power an user-facing feature – [Rokt](http://rokt.com), an e-commerce platform, uses ClickHouse to power its analytics panels. However, some companies leverage ClickHouse for internal use cases, such as the Washington Post, which uses ClickHouse to power its in-house analytics suite. 
+
+## Analytics performance comparison
+
+ClickHouse was built to perform aggregations, but it’s naive to say that Elasticsearch doesn’t have the structure to compete with ClickHouse on *some* aggregations. 
+
+To understand this, remember the primary philosophy behind ClickHouse’s design: pre-calculate aggregations ahead of queries to enable millisecond-level fetches. ClickHouse accomplishes this through materialized views and specialized engines, which are optimized for mathematical queries transversing numeric data. 
+
+Elasticsearch, meanwhile, can accomplish similar performance over certain queries. For instance, if a product needs to print the number of college alumni that are unemployed, Elasticsearch can accomplish this by adding up indices in the inverted index of words that pattern-match to unemployed alumni. In other words, fast search sometimes equates to great analytics by just adding a COUNT() function. 
+
+While both Elasticsearch and ClickHouse are fundamentally backend products, we can compare their respective GUI products – Kibana for Elasticsearch and ClickHouse Cloud for ClickHouse. ClickHouse Cloud is a *much* younger product; Kibana, conversely, has been around for nearly a decade and has an extensive UI.
+
+![Comparinson.png](../images/blog/clickhouse-vs-elastic/comparison.png)
+
+In a nutshell, comparing the analytics efficiency of ClickHouse and Elasticsearch has the same sort-of, not-really awkwardness of other comparisons – they both excel in their respective categories using radically different methods, just catering to a different type of need. However, Elastic’s Kibana product is more mature than ClickHouse Cloud’s competitive offering. 
+
+## Comparing developers
+
+One of the reasons that Elasticsearch grew quickly thanks to fast-paced development by its parent company, Elastic. Elastic has spearheaded the development of Elasticsearch, Kibana, and other accessory products like Beats, a data shipper. For many enterprise customers, Elastic being a psuedo-closed-source solution is balanced by the fact that it can leverage its enterprise revenue to foster a massive engineering effort to improve Elastic.
+
+Conversely, the teams behind ClickHouse are a lot smaller. However, ClickHouse has a very avid developer community, with contributors existing outside of the two major ClickHouse developers – ClickHouse Inc and Altinity Inc. And one of the reasons that ClickHouse is starting to really grow in the last few years is because of its open-source, pro-community brand. 
```

Comment:
```suggestion
Conversely, the teams behind ClickHouse are a lot smaller. However, ClickHouse has a very avid developer community, with contributors existing outside of the two major ClickHouse developers – ClickHouse Inc and Altinity Inc And one of the reasons that ClickHouse is starting to really grow in the last few years is because of its open-source, pro-community brand. 
```
If we want a period after Inc we should do so everywhere 😅

Reply:
Ah, my bad. I moved this into the conclusion but forgot to delete the previous section.